entities/storobj/storage_object.go

```diff
@@ -432,26 +434,80 @@ func (ko *Object) MarshalBinary() ([]byte, error) {
 // UnmarshalPropertiesFromObject only unmarshals and returns the properties part of the object
 //
 // Check MarshalBinary for the order of elements in the input array
-func UnmarshalPropertiesFromObject(data []byte, properties *models.PropertySchema) error {
+func UnmarshalPropertiesFromObject(data []byte, properties *map[string]interface{}, aggregationProperties []string, propStrings [][]string) error {
 	if data[0] != uint8(1) {
 		return errors.Errorf("unsupported binary marshaller version %d", data[0])
 	}
 
+	// clear out old values in case an object misses values. This should NOT shrink the capacity of the map, eg there
+	// are no allocations when adding the properties of the next object again
+	for k := range *properties {
+		delete(*properties, k)
+	}
+
 	startPos := uint64(1 + 8 + 1 + 16 + 8 + 8) // elements at the start
 	byteOps := byte_operations.ByteOperations{Position: startPos, Buffer: data}
 	// get the length of the vector, each element is a float32 (4 bytes)
 	vectorLength := uint64(byteOps.ReadUint16())
 	byteOps.MoveBufferPositionForward(vectorLength * 4)
 
-	// length of class name
 	classnameLength := uint64(byteOps.ReadUint16())
 	byteOps.MoveBufferPositionForward(classnameLength)
-
-	// property schema length
 	propertyLength := uint64(byteOps.ReadUint32())
-	if err := json.Unmarshal(data[byteOps.Position:byteOps.Position+propertyLength], properties); err != nil {
-		return err
-	}
+
+	jsonparser.EachKey(data[byteOps.Position:byteOps.Position+propertyLength], func(idx int, value []byte, dataType jsonparser.ValueType, err error) {
+		switch dataType {
+		case jsonparser.Number:
+			v, _ := jsonparser.ParseFloat(value)
```

Comment:
Why are you here ignoring the returned `err` but when you parse array types then you're not ignoring it

Reply:
added errorchecking for those